### PR TITLE
fix(query): fix function if result incorrect bug

### LINKED ISSUE #7232: `IF` function produce wrong result when input is NULL
```sql
create table t(a int null, b int);
insert into t values(null, 1);

select a, b, a*2, if(b=1, a*2, b) from t; -- expected to be NULL, 1, NULL, NULL
+------+------+-------+---------------------+
| a    | b    | a * 2 | if(b = 1, a * 2, b) |
+------+------+-------+---------------------+
| NULL |    1 |  NULL |                   0 |
+------+------+-------+---------------------+

select a, b, a*2, if(b=1, a, b) from t;
+------+------+-------+-----------------+
| a    | b    | a * 2 | if(b = 1, a, b) |
+------+------+-------+-----------------+
| NULL |    1 |  NULL |            NULL |
+------+------+-------+-----------------+

select a, b, a*2, if(1, a*2, b) from t;
+------+------+-------+-----------------+
| a    | b    | a * 2 | if(1, a * 2, b) |
+------+------+-------+-----------------+
| NULL |    1 |  NULL |            NULL |
+------+------+-------+-----------------+
```

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

1. Fix function if result incorrect bug
2. Replace `std::alloc::allocate` with Global API.

Fixes #7232

### CODE REVIEW

**Review by leiysky — APPROVED** [review]

File: tests/logictest/suites/base/02_function/02_0010_function_if [hunk 1]
```diff
@@ -103,3 +103,20 @@ select if(true, number, null), if(false, number, null) from numbers(1);
 ----
 0 NULL
 
+statement ok
```
Comment (leiysky) [inline tests/logictest/suites/base/02_function/02_0010_function_if:106]:
```suggestion
statement ok
drop table if exists t;

statement ok
```

**Review by PsiACE — APPROVED** [review]

**Comment by leiysky** [comment]
@mergify update

**Comment by Xuanwo** [comment]
test_sqllogic_ydb_standalone_linux failed for timeout.

File: tests/logictest/complete.py [hunk 2]
```diff
@@ -121,9 +137,14 @@ def gen_suite_from_sql(sqls, dest_file):
...
-            statements.append(
-                f"statement query {options}\n{sql}\n\n----\n{format_result(results)}\n"
-            )
+            if len(results) > 0:
+                statements.append(
+                    f"statement query {options}\n{sql}\n\n----\n{format_result(results)}\n"
+                )
+            else:
+                statements.append(
+                    f"statement ok\n{sql}\n"
```
Comment (ZeaLoVe) [inline tests/logictest/complete.py:146]: statement ok need two '\n' in the end.

File: tests/logictest/complete.py [hunk 3]
```diff
@@ -121,9 +137,14 @@ def gen_suite_from_sql(sqls, dest_file):
...
+                )
         except mysql.connector.Error as err:
             statements.append(f"statement ok\n{sql}\n\n")
```
Comment (ZeaLoVe) [inline tests/logictest/complete.py:149]: better change to `statements.append(f"statement error {err.errno}\n{sql}\n\n")`

Comment (sundy-li) [reply tests/logictest/complete.py:149]: It's `No result set to fetch from.`

Comment (ZeaLoVe) [reply tests/logictest/complete.py:149]: > It's `No result set to fetch from.`

Fine, that is a statement ok

**Review by ZeaLoVe — APPROVED** [review]

**Pull request merged**